pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/AggregationOperator.java

```diff
@@ -97,15 +69,12 @@ public Block getNextBlock(BlockId blockId) {
             _numTotalRawDocs);
 
     // Build intermediate result block based on aggregation result from the executor.
-    List<Serializable> aggregationResults = _aggregationExecutor.getResult();
-    return new IntermediateResultsBlock(AggregationFunctionFactory.getAggregationFunction(_aggregationInfoList),
-        aggregationResults);
+    return new IntermediateResultsBlock(_aggregationFunctionContexts, aggregationExecutor.getResult(), false);
   }
 
   @Override
-  public boolean open() {
-    _projectionOperator.open();
-    return true;
+  public Block getNextBlock(BlockId blockId) {
```

Comment:
better to keep the behavior symmetric across all operators, just check for blockId = 0 and return null for everything else?